pkg/cmd/server/kubernetes/node/node_config.go

```diff
@@ -237,8 +237,19 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig, enableProxy, enable
 	internalKubeInformers := kinternalinformers.NewSharedInformerFactory(kubeClient, proxyconfig.ConfigSyncPeriod)
 
 	// Initialize SDN before building kubelet config so it can modify option
-	sdnPlugin, err := sdnplugin.NewNodePlugin(options.NetworkConfig.NetworkPluginName, originClient, kubeClient, internalKubeInformers, options.NodeName, options.NodeIP,
-		options.NetworkConfig.MTU, proxyconfig.KubeProxyConfiguration, options.DockerConfig.DockerShimSocket)
+	sdnNodeConfig := &sdnplugin.OsdnNodeConfig{
+		PluginName:         options.NetworkConfig.NetworkPluginName,
+		Hostname:           options.NodeName,
+		SelfIP:             options.NodeIP,
+		RuntimeEndpoint:    options.DockerConfig.DockerShimSocket,
+		MTU:                options.NetworkConfig.MTU,
+		OSClient:           originClient,
+		KClient:            kubeClient,
+		KubeInformers:      internalKubeInformers,
```

Comment:
hm... missed this before: osClient, kClient, and kubeInformers aren't really "configuration" like the rest of the stuff. Maybe keep them still as separate arguments